os/binfmt/libelf/libelf_bind.c

```diff
@@ -96,25 +96,57 @@ static int g_num_lib_syms;
  * Private Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: elf_enable_caching
+ *
+ * Description:
+ *   Enable caching for large tables during relocation.
+ *   This API is to be called only if we are not able to allocate the complete
+ *   table (symtab, reltab, strtab) due to memory limitation.
+ *
+ *   Even if there is a failure to allocate cache, we can still try
+ *   to load directly from the flash or file system. This might take
+ *   too much time depending on binary size and number of relocations.
+ *
+ *
+ ****************************************************************************/
+static inline void elf_enable_caching(FAR struct elf_loadinfo_s *loadinfo) {
+#if defined(CONFIG_ELF_CACHE_READ)
+	if (elf_cache_init(loadinfo->filfd, loadinfo->offset, loadinfo->filelen) != OK) {
```

Comment:
How about checking `cached_read` first?
Because `elf_enable_caching` is called from many where, many times.
So if we check the `cached_read` first, we can skip to init again.
```
if (!loadinfo->cached_read && elf_cache_init(loadinfo->filfd, loadinfo->offset, loadinfo->filelen) != OK) {
```

Reply:
Done. Thanks.

Reply:
You left `Done`, but there is no checking for `cache_read`.
Could you check again?

Reply:
I have added the check at the point of reading the tables. I feel this is better. 
Like... `if (!loadinfo->cached_read && elf_readstrtab(loadinfo) == -ENOMEM)`
For example.. if there is failure while reading symtab.. Then we allocate cache.
In this case, we dont need to try for reading strtab and reltab. We will directly use the cache.

Reply:
Okay. Checked. Thanks.